virtcontainers/device/manager/utils.go

```diff
@@ -35,3 +40,92 @@ func isVFIO(hostPath string) bool {
 func isBlock(devInfo config.DeviceInfo) bool {
 	return devInfo.DevType == "b"
 }
+
+func IsVFIOLargeBarSpaceDevice(hostPath string) (bool, error) {
+	if !isVFIO(hostPath) {
+		return false, nil
+	}
+	iommuDevicesPath := filepath.Join(config.SysIOMMUPath, filepath.Base(hostPath), "devices")
+	deviceFiles, err := ioutil.ReadDir(iommuDevicesPath)
+	if err != nil {
+		return false, err
+	}
+	// Pass all devices in iommu group
+	for _, deviceFile := range deviceFiles {
+		//For example, 0000:04:00.0
+		tokens := strings.Split(deviceFile.Name(), ":")
+		vfioDeviceType := config.VFIODeviceErrorType
+		if len(tokens) == 3 {
+			vfioDeviceType = config.VFIODeviceNormalType
+		} else {
+			//For example, 83b8f4f2-509f-382f-3c1e-e6bfe0fa1001
+			tokens = strings.Split(deviceFile.Name(), "-")
+			if len(tokens) == 5 {
+				vfioDeviceType = config.VFIODeviceMediatedType
+			}
```

Comment:
Should be not error out or return here ?

Reply:
becase `case config.VFIODeviceMediatedType` is TODO.